wrappers/ios/Sources/Wrapper/Reader/ZXIBarcodeReader.mm

```diff
@@ -115,16 +147,22 @@ + (DecodeHints)DecodeHintsFromZXIOptions:(ZXIDecodeHints*)hints {
 
     NSMutableArray* zxiResults = [NSMutableArray array];
     for (auto result: results) {
-        auto resultText = result.text();
-        NSString *text = [[NSString alloc]initWithBytes:resultText.data() length:resultText.size() encoding:NSUTF8StringEncoding];
-
-        NSData *bytes = [[NSData alloc] initWithBytes:result.bytes().data() length:result.bytes().size()];
         [zxiResults addObject:
-         [[ZXIResult alloc] init:text
+         [[ZXIResult alloc] init:stringToNSString(result.text())
                           format:ZXIFormatFromBarcodeFormat(result.format())
-                           bytes:bytes
+                           bytes:[[NSData alloc] initWithBytes:result.bytes().data() length:result.bytes().size()]
                         position:[[ZXIPosition alloc]initWithPosition: result.position()]
-         ]];
+                     orientation:result.orientation()
+                         ecLevel:stringToNSString(result.ecLevel())
+             symbologyIdentifier:stringToNSString(result.symbologyIdentifier())
+                    sequenceSize:result.sequenceSize()
+                   sequenceIndex:result.sequenceIndex()
+                      sequenceId:stringToNSString(result.sequenceId())
+                      readerInit:result.readerInit()
+                       lineCount:result.lineCount()
+                         version:stringToNSString(result.version())
```

Comment:
Please remove `version`. I intend to deprecate it because it currently is an inconsistent mix of size and version info where only size (in modules) is truly meaningful for all symbologies, whereas the term "version" really only exists for QRCodes.

Reply:
`version` is now removed in 18c3fef00ad657605a301e06489b74415d978fe2